enterprise/pkg/a8n/resolvers/campaign_plans.go

```diff
@@ -56,34 +58,277 @@ func (r *campaignPlanResolver) Changesets(
+func (r *campaignJobsConnectionResolver) PageInfo(ctx context.Context) (*graphqlutil.PageInfo, error) {
+	_, _, next, err := r.compute(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return graphqlutil.HasNextPage(next != 0), nil
+}
+
+type campaignJobResolver struct {
+	job           *a8n.CampaignJob
+	preloadedRepo *repos.Repo
+
+	// cache repo because it's called more than one time
+	once   sync.Once
+	repo   *graphqlbackend.RepositoryResolver
+	commit *graphqlbackend.GitCommitResolver
+	err    error
+}
+
+func (r *campaignJobResolver) Title() (string, error) { return "Title placeholder", nil }
+func (r *campaignJobResolver) Body() (string, error)  { return "Body placeholder", nil }
+
+func (r *campaignJobResolver) computeRepoCommit(ctx context.Context) (*graphqlbackend.RepositoryResolver, *graphqlbackend.GitCommitResolver, error) {
+	r.once.Do(func() {
+		if r.preloadedRepo != nil {
+			r.repo = newRepositoryResolver(r.preloadedRepo)
+		} else {
+			r.repo, r.err = graphqlbackend.RepositoryByIDInt32(ctx, api.RepoID(r.job.RepoID))
+			if r.err != nil {
+				return
+			}
+		}
+		args := &graphqlbackend.RepositoryCommitArgs{Rev: string(r.job.Rev)}
+		r.commit, r.err = r.repo.Commit(ctx, args)
+	})
+	return r.repo, r.commit, r.err
+}
+
+func (r *campaignJobResolver) Repository(ctx context.Context) (*graphqlbackend.RepositoryResolver, error) {
+	repo, _, err := r.computeRepoCommit(ctx)
+	return repo, err
+}
+
+func (r *campaignJobResolver) BaseRepository(ctx context.Context) (*graphqlbackend.RepositoryResolver, error) {
+	repo, _, err := r.computeRepoCommit(ctx)
+	return repo, err
+}
+
+func (r *campaignJobResolver) Diff(ctx context.Context) graphqlbackend.ChangesetPlanResolver {
+	return r
+}
+
+func (r *campaignJobResolver) FileDiffs(ctx context.Context, args *graphqlutil.ConnectionArgs) (graphqlbackend.PreviewFileDiffConnection, error) {
+	_, commit, err := r.computeRepoCommit(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &previewFileDiffConnectionResolver{
+		job:    r.job,
+		commit: commit,
+		first:  args.First,
+	}, nil
+}
+
+type previewFileDiffConnectionResolver struct {
+	job    *a8n.CampaignJob
+	commit *graphqlbackend.GitCommitResolver
+	first  *int32
+
+	// cache result because it is used by multiple fields
+	once        sync.Once
+	fileDiffs   []*diff.FileDiff
+	hasNextPage bool
+	err         error
+}
+
+func (r *previewFileDiffConnectionResolver) compute(ctx context.Context) ([]*diff.FileDiff, error) {
+	r.once.Do(func() {
+		r.fileDiffs, r.err = diff.ParseMultiFileDiff([]byte(r.job.Diff))
+		if r.err != nil {
+			return
+		}
+
+		if r.first != nil && len(r.fileDiffs) > int(*r.first) {
+			r.hasNextPage = true
+		}
+	})
+	return r.fileDiffs, r.err
+}
+
+func (r *previewFileDiffConnectionResolver) Nodes(ctx context.Context) ([]graphqlbackend.PreviewFileDiff, error) {
+	fileDiffs, err := r.compute(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.first != nil && len(fileDiffs) > int(*r.first) {
```

Comment:
I find this easier to read. Just a suggestion.

```suggestion
	if r.first != nil && int(*r.first) <= len(fileDiffs) {
```